### PR TITLE
Initial implementation of the interpolated legend

### PR DESCRIPTION
**Pull request by Ro4052**
An initial implementation of the new legend put up for code review so you guys can see how I've approached it.

### CODE REVIEW

File: packages/perspective-viewer-d3fc/src/js/legend/interpolatedLegend.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        .append("svg")
+        .style("width", width)
+        .style("height", height);
+    const legendBar = legendSvg
+        .append("g")
+        .datum(expandedDomain)
+        .call(svgBar);
+
+    const barWidth = Math.abs(legendBar.node().getBBox().x);
+    legendSvg
```
Comment (andy-lee-eng) [inline packages/perspective-viewer-d3fc/src/js/legend/interpolatedLegend.js]: Are you adding two `svg` elements here? Should be able to add both `g` elements to the same `svg`

**Comment by andy-lee-eng** [comment]
Looks good, and the code should drop into the d3-component style version that I've just changed the other legend to.

e.g.:

````javascript
export function colourRangeLegend() {
    let scale = null;

    function legend(container) {
        // get the domain from the scale
        const domain = scale.domain();

        // ... rest of rendering code
    }

    legend.scale = (...args) => {
        if (!args.length) {
            return scale;
        }
        scale = args[0];
        return legend;
    };

    return legend;
}

// Rendering the legend (in the chart)
const legend = colourRangeLegend().scale(someColourScale);
container.call(legend);
````

**Review by mhooper-scottlogic — COMMENTED** [review]
Looks good.

We need to decide what we are going to do about the label text:
image

The tricky thing is that it will depend on the theme background and whether or not the bit of the grid where the legend is printed is coloured.

When the range of values is very low, the legend gets all squashed in the middle:
image

**Comment by andy-lee-eng** [comment]
For Heatmap (which always draws in the top-left), maybe we should just put some right-padding on the chart so that it doesn't overlap the legend?

````
    .chart {

        &.d3_heatmap {
            padding-right: 165px;
        }

    }
````

**Comment by Ro4052** [comment]
@matt-hooper, hopefully the issues you've raised should be fixed now. For the first one, the light and dark themes should now both be supported. As for the second, I was padding the data by 1 on either side for any data set, this padding is now a percentage instead.

**Review by andy-lee-eng — APPROVED** [review]
Looks good.

I think it looks quite nice if you hide the axis domain line (but leave the ticks). Something like this:

````javascript
            .call(axisLabel)
            .select(".domain")
            .attr("visibility", "hidden");
````

**Comment by Ro4052** [comment]
I was wondering about something like this, I'll sort it out now.

**Pull request merged**